src/Adapter/DynamoDbFixtureTrait.php

```diff
@@ -0,0 +1,170 @@
+            ]);
+        } catch (DynamoDbException $e) {
+            if ($throwOnFail) {
+                throw new LoadFailedException(
+                    sprintf('Failed to load single record to table "%s": %s', $this->getTableName(), $e->getMessage()),
+                    [$e->toArray()],
+                );
+            }
+        }
+    }
+
+    /**
+     * @param array<int, Record> $records
+     */
+    protected function loadBatchRecords(DynamoDbClient $dynamoDb, array $records, bool $throwOnFail = true): void
+    {
+        if (empty($records)) {
+            return;
+        }
+
+        // DynamoDB batch write supports up to 25 items per request
+        $chunks = array_chunk($records, 25);
+
+        foreach ($chunks as $chunk) {
+            $requestItems = [];
+            foreach ($chunk as $record) {
+                $requestItems[] = [
+                    'PutRequest' => [
+                        'Item' => $record->getValues(),
+                    ],
+                ];
+            }
+
+            try {
+                $result = $dynamoDb->batchWriteItem([
+                    'RequestItems' => [
+                        $this->getTableName() => $requestItems,
+                    ],
+                ]);
+
+                // Handle unprocessed items
+                if (!empty($result['UnprocessedItems'])) {
+                    $this->handleUnprocessedItems($dynamoDb, $result['UnprocessedItems'], $throwOnFail);
+                }
+            } catch (DynamoDbException $e) {
+                if ($throwOnFail) {
+                    throw new LoadFailedException(
+                        sprintf('Failed to load batch records to table "%s": %s', $this->getTableName(), $e->getMessage()),
+                        [$e->toArray()],
+                    );
+                }
+            }
+        }
+    }
+
+    private function handleUnprocessedItems(DynamoDbClient $dynamoDb, array $unprocessedItems, bool $throwOnFail): void
+    {
+        $maxRetries = 3;
+        $retryCount = 0;
+
+        while (!empty($unprocessedItems) && $retryCount < $maxRetries) {
+            ++$retryCount;
+
+            // Exponential backoff
+            usleep(100000 * $retryCount); // 100ms, 200ms, 300ms
+
+            try {
+                $result = $dynamoDb->batchWriteItem([
+                    'RequestItems' => $unprocessedItems,
+                ]);
+
+                $unprocessedItems = $result['UnprocessedItems'] ?? [];
+            } catch (DynamoDbException $e) {
+                if ($throwOnFail) {
+                    throw new LoadFailedException(
+                        sprintf('Failed to process unprocessed items for table "%s": %s', $this->getTableName(), $e->getMessage()),
+                        [$e->toArray()],
+                    );
+                }
+                break;
+            }
+        }
+
+        if (!empty($unprocessedItems) && $throwOnFail) {
+            throw new LoadFailedException(
```

Comment:
Split the sprintf accross several lines to avoid extra long lines

```php
            throw new LoadFailedException(
                sprintf(
                    'Failed to process all items for table "%s" after %d retries',
                    $this->getTableName(),
                    $maxRetries
                ),
                [$unprocessedItems],
            );
```